CMakeLists.txt

```diff
@@ -0,0 +1,88 @@
+cmake_minimum_required(VERSION 2.8)
+
+set(CMAKE_TOOLCHAIN_FILE "${CMAKE_SOURCE_DIR}/generic-msp430-gcc.cmake")
+
+# Name your project
+project(EXAMPLE)
+
+#device dependent config, here for launchpad
+set(MSP430_MCU msp430fr2355)
+set(MSP430_MCU_FREQ 24000000UL)
+
+set(CMAKE_C_STANDARD 11)
+set(CMAKE_C_STANDARD_REQUIRED ON)
+# set(CMAKE_C_EXTENSIONS ON)
+
+add_compile_options(-Werror -ffunction-sections -fdata-sections -mmcu=${MSP430_MCU} -DF_CPU=${MSP430_MCU_FREQ})
+
+
+# note-c
+set(NOTE_C ${CMAKE_CURRENT_LIST_DIR}/note-c)
+add_library(note-c OBJECT
+    ${NOTE_C}/n_atof.c
+    ${NOTE_C}/n_b64.c
+    ${NOTE_C}/n_cjson.c
+    ${NOTE_C}/n_cjson_helpers.c
+    ${NOTE_C}/n_const.c
+    ${NOTE_C}/n_ftoa.c
+    ${NOTE_C}/n_helpers.c
+    ${NOTE_C}/n_hooks.c
+    ${NOTE_C}/n_i2c.c
+#    ${NOTE_C}/n_md5.c
+    ${NOTE_C}/n_request.c
+    ${NOTE_C}/n_serial.c
+    ${NOTE_C}/n_str.c
+#    ${NOTE_C}/n_ua.c
+)
+target_include_directories(note-c
+    PUBLIC ${NOTE_C}
+)
+
+# driverlib
+cmake_path(SET DRIVERLIB NORMALIZE "${CMAKE_CURRENT_LIST_DIR}/driverlib/MSP430FR2xx_4xx")
+add_library(driverlib OBJECT
+    ${DRIVERLIB}/cs.c
+    ${DRIVERLIB}/eusci_b_i2c.c
+    ${DRIVERLIB}/eusci_a_uart.c
+    ${DRIVERLIB}/framctl.c
+    ${DRIVERLIB}/gpio.c
+    ${DRIVERLIB}/pmm.c
+    ${DRIVERLIB}/timer_b.c
+    ${DRIVERLIB}/wdt_a.c
+)
+
+target_include_directories(driverlib PUBLIC "${DRIVERLIB}" "${MSP430_GCC_PATH}/include")
+
+function(msp430_add_executable_and_dependencies EXECUTABLE)
+	set(EXECUTABLE_ELF "${EXECUTABLE}.elf")
+    msp430_add_executable(${EXECUTABLE} main.c ${ARGN})
+    # include the source root for main.h
+    target_include_directories(${EXECUTABLE_ELF} PUBLIC "./")
```

Comment:
Interesting -- does this not happen by default? I would've thought the current directory is automatically in the include path, but maybe not.

Reply:
Oh, I see the indentation problem above. :-)

Reply:
Removed the include of the current folder. At one point the build wasn't finding `main.h`.